modules/core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestColdStorage.java

```diff
@@ -244,6 +244,43 @@ public void shouldCheckAvailability() {
         checkAvailabilityOfDocuments(Arrays.asList(documents.get(1), documents.get(2)), downloadableUntil, 0);
     }
 
+    @Test
+    public void shouldFailUpdateMainContentAlreadyInColdStorage() throws IOException {
+        // move the main content into the cold storage
+        DocumentModel documentModel = createFileDocument(DEFAULT_DOC_NAME, true);
+        moveAndVerifyContent(session, documentModel);
+
+        // we cannot update the main content as it is already in cold storage
+        documentModel = session.getDocument(documentModel.getRef());
```

Comment:
Don't we want to use `DocumentModel#refresh` now @efge ?

Reply:
Yes it's shorter and more explicit but no big deal.